library/alloc/src/string.rs

```diff
@@ -1593,6 +1593,12 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    #[unstable(feature = "liballoc_internals", issue = "none", reason = "implementation detail")]
+    #[doc(hidden)]
+    pub fn __push_fmt(&mut self, args: fmt::Arguments<'_>) {
```

Comment:
Why can't it just be an inherent `write_fmt` method, so write!(...) always works on String instead of introducing a semi-hidden method and a whole new macro?

Reply:
That won’t be backward compatible: write_fmt already exists on String (via Write trait) and returns a result.

Reply:
Well I guess it would have to have the same signature. So I guess the unwrapping would unfortunately stay in that case.